### DISCUSSION TITLE
GPS Module Support?

### DISCUSSION THREAD

**Discussion by mlevett** [richonguzman/LoRa_APRS_iGate#131] [Q&A]
Good evening,  I'm looking to use this for my own igate/digi but was wondering if there is support for a GPS module to automatically apply the Lat and Long plus accurate time etc? 

Thanks,

**Comment by richonguzman** [comment]
Hi, 

soon I will add the possibility to get its own position from GPS at boot up

or are you planing on having a mobile igate?

**Reply by g3cwi** [reply]
+1 for this. I use my igates in a portable role. They are brilliant for setting up temporary/emergancy networks and being able to use the GPS position would be very desirable. Thanks.

**Reply by knutsnilsen-max** [reply]
Hi, could we expect this implemented soon?

**Reply by richonguzman** [reply]
> soon

@knutsnilsen-max  this has been working for many months now. Check the Wiki

**Comment by mlevett** [comment]
Thanks!  I was planning for it do pick it up on boot and have accurate location and time.  May also use it mobile now and again but would likely be deployed at a static location out in the field. I'm also looking to build your tracker too, just need to work out what pins to connect to where to add a GPS module to the Heltec V3 boards I have.On 1 Jul 2024 1:21 am, "Ricardo Guzman (Richonguzman)" ***@***.***> wrote:
Hi,
soon I will add the possibility to get its own position from GPS at boot up
or are you planing on having a mobile igate?

**Reply by arizonajon1** [reply]
have you figured out which pins to use for a serial GPS receiver? My desire is a mobile operation, so the GPS position will be constantly updating.

**Comment by SA7SKY** [comment]
Please all, be reminded that fixed stations that transmit a GPS location will, by the fluctuation of the postion +/- a few decimal degrees, be regarded by i.e. aprs.fi as mobile stations.
I always found that annoying and meanless.
So, GPS for homebases, igates, digipeaters, etc. are not recommended IMHO.
Other than that: It 's a free world...

**Reply by arizonajon1** [reply]
Another interesting thing - at least using the CA2RXU firmware with a gps-equipped T-beam or Supreme, the slow "wobbling" you discuss even when the device is being used in fixed location is enough that the "WX station while fixed" feature never activates. %^(

**Reply by richonguzman** [reply]
@arizonajon1 this has been changed now, telemetry replaces de Wx packets. As moving stations surely wont tx any real Wx station packet, but its data could be obverved as real telemetry to be used in another way than regular Wx Packet from standardized stations

**Comment by mlevett** [comment]
Agree with you there, what I personally think would be good is for it to get a fix on boot up and then lock to that location. Then it can "auto location" it's self and then if once it's got a lock it does not update again so as to avoid the jitter issue.All IMO of course.  My use case is so I could take it out with me and power it up and it just works. No faffing about finding location, then updating the config etc. On 1 Jul 2024 9:55 am, Helge SA7SKY ***@***.***> wrote:
Please all, be reminded that fixed stations that transmit a GPS location will, by the fluctuation of the postion +/- a few decimal degrees, be regarded by i.e. aprs.fi as mobile stations.
I always found that annoying and meanless.
So, GPS for homebases, igates, digipeaters, etc. are not recommended IMHO.
Other than that: It 's a free world...

**Reply by SA7SKY** [reply]
That sounds like a clever idea with the 'start-up, first position, log-it' for the session until shut down.
I am not a programmer but I wonder about the date/time which is taken from the GNSS as well.
So, telling the system to "disregard" further position updates (NMEA) but accept the time updates a the same time could be tough.
On the other hand APRS-IS puts a timestamp upon reception anyway.
But then perhaps some BME modules need time/date data for a complete string of data output.
That is out of my scope.
So, we leave it to the hands of the coder ;-)
Thanks for the idea though.
There is always something new to consider!
73

**Reply by g3cwi** [reply]
Agreed about having the location "fixed" although some people have used mobile igates (that are actually moving)...

**Comment by Grahamw99** [comment]
Hi all, 
Please can someone advise if GPS support was ever added. Similar to some comments above, I intend to use a mobile igate in my van whilst I am out SOTAing. Therefore GPS support that fixes a location on boot and is then ignored, would be ideal.

**Reply by richonguzman** [reply]
Hi, for more than a year boards with GPS can use it for a mobile Station/iGate 

do you plan to only get one position and stay with it for the time is being powered (until next reboot)?

**Reply by arizonajon1** [reply]
> Hi all, Please can someone advise if GPS support was ever added. Similar to some comments above, I intend to use a mobile igate in my van whilst I am out SOTAing. Therefore GPS support that fixes a location on boot and is then ignored, would be ideal.

Hi there - While I have never done SOTA, my situation and needs may be somewhat similar.
 
I use a LilyGO Supreme that's permanently mounted in the truck connected to the truck GPS and is my 70 cm LoRa truck tracker. It has a 1 watt amp for extended range. The Supreme is also set up to be a WIDE1-1 digipeater. When I carry my handheld T-Deck on a hike or go into a shop, that WIDE1-1 hop really helps me to stay in touch. When parked, the truck digi's position report rate drops significantly. And if, the truck actually gets stolen (never happened, but I suppose it could) at least I'd know where it was %^)

All my LoRa gear uses the CA2RXU firmware.

Cheers and 73 - Jon N7UV

**Reply by Grahamw99** [reply]
> Hi, for more than a year boards with GPS can use it for a mobile Station/iGate
> 
> do you plan to only get one position and stay with it for the time is being powered (until next reboot)?

Hi.
Thanks for the reply. Yes I looked at the set up again and discovered the option to have "real GPS Position" reported. I have enabled this and intend adding a GPS module to the board.
Yes, I plan to get to one position, boot up the igate and stay there until I need to move. The igate will be switched off until the next location is reached several days later.

**Reply by richonguzman** [reply]
note: the GPS process is only for boards with onboard GPS (and also LoRa32 + GPS (DIY))

which board do you plan to add GPS ? to check of how to achieve this.

**Reply by Grahamw99** [reply]
I have a Heltec V3 which seems to be working fine. I intend to add on a NEO6MU2 GPs module.

**Reply by richonguzman** [reply]
> I have a Heltec V3 which seems to be working fine. I intend to add on a NEO6MU2 GPs module.

what is the baud rate of this unit? does it need only RX and TX connections (besides VCC and GND ? )?

**Reply by Grahamw99** [reply]
The baud rate is 9600 and it needs all 4 connection VCC, GND, RX & TX
![Screenshot_20260401_061803_DuckDuckGo](https://github.com/user-attachments/assets/d250c377-cfe8-4bbf-8ad0-080af8e0268a)

**Reply by richonguzman** [reply]
so it should work right away...

will add it for all Heltecs V3  next week

**Reply by Grahamw99** [reply]
Great, Thanks.
